### ISSUE TITLE
How concatenate an embedding with outputs of LSTM with variable length

### ISSUE THREAD

**Issue by Liuch10** [keras-team/keras#7582]
i've got an LSTM with _return_sequence =True_ and _mast_zero=True_, whose output shape _(Samples, Timesteps, units)= (None, None, output_size)_ because of the variable timesteps.

Furthermore, I want to concatenate an embedding with each of the output of LSTM, to merge the information from another data modality.

However, the _Timesteps_ of  Embedding layers can't be None(i.e. it can't be variable) even when I user mask_zeo.

How can i do to fix this?

**Comment by samre12** [comment]
Is there any upper limit on the number of timesteps that you consider?
If there is, then you could do something like:
- pad the input to the maximum number of time steps (denote it by `max_timesteps`) and then pass it to embeddings layer 
- then you could use `Keras.backend.gather` to obtain the first valid timesteps of input
- then use `Keras.backend.concatenate`

**Comment by Liuch10** [comment]
@samre12  thanks bro. 
Yes indeed I've got a %max_timesteps%, do you mean that i will pass the output_mask of the previous Embedding layer and use gather to truncate the tensor?
However, Keras.backend.gather seems only perform such operation over the first dimension(which is the sample dim)
  ```python
      # Scalar indices
      output[:, ..., :] = params[indices, :, ... :]

      # Vector indices
      output[i, :, ..., :] = params[indices[i], :, ... :]

      # Higher rank indices
      output[i, ..., j, :, ... :] = params[indices[i, ..., j], :, ..., :]
  ```

**Comment by samre12** [comment]
@Liuch10 , thanks for pointing that out. You could use `keras.backend.permute_dimensions` with `keras.backend.gather` to make it work.

**Comment by paulx3** [comment]
@samre12 Hi, I met the same problem in implementing VAE with condition. Could you please write an example on how to use these two functions? Thanks
